packages/@aws-cdk/cdk/lib/core/tag-manager.ts

```diff
@@ -0,0 +1,149 @@
+import { Construct } from './construct';
+import { Token } from './tokens';
+
+/**
+ * ITaggable indicates a entity manages tags via the `tags` property
+ */
+export interface ITaggable {
+    tags: TagManager,
+}
+
+/**
+ * A single tag property values
+ */
+interface ITagProps {
+    value: string | Token;
+    propagate?: boolean;
+}
+
+/**
+ * A an object of tags
+ */
+export interface ITags {
+    [key: string]: ITagProps;
```

Comment:
Not sure this is needed

Reply:
I modified this to be concrete struct to ensure that properties were set internal to the class. I removed the export.